staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchPathProvider.java

```diff
@@ -109,6 +126,68 @@ public Path getTenantInprogressDirectory(final TenantId tenantId)
         return getPathForTenant(tenantId).resolve(INPROGRESS_FOLDER);
     }
 
+    public void monitorBatchProgressInFileSystem(final TenantId tenantId, final BatchId batchId, String threadID)
+    {
+        final Path pathForBatches = getTenantInprogressDirectory(tenantId);
+        final long[] size = new long[1];
+        try (Stream<Path> fileStream = Files.list(pathForBatches)) {
+            fileStream.filter(batch -> BatchNameProvider.getBatchId(batch.getFileName().toString()).equals(batchId.getValue()) &&
+                            !batch.getFileName().toString().contains(ServiceIdentifier.getServiceId()))
+                    .map(batch -> batch.resolve("files"))
+                    .forEach(batch -> {
+                        try (Stream<Path> filePath = Files.walk(batch)) {
+                            size[0] = filePath.mapToLong(file -> file.toFile().length()).sum();
+                        } catch (IOException e) {
+                            LOGGER.info("Folder might have been deleted or moved.");
```

Comment:
Right, if the batch is completed during parsing, we will get IOException that file not found. I am now logging this in error log level.

Reply:
This was being caught in various places so I adjusted the code in commit 590f331c0db34ae520d5a2d8a635c62a07f8bd09 to let it bubble up so that it can be handled in the main `getStatus()` method.